File: loader/src/main/java/moe/caa/multilogin/loader/library/Library.java
package moe.caa.multilogin.loader.library;

import java.io.File;
import java.util.List;

public record Library(
        String group,
        String name,
        String version
) {
    public static Library of(String str) {
        String[] strings = str.split(":");
        return new Library(strings[0], strings[1], strings[2]);
    }

    public File getFile(File folder) {
        return new File(folder, getUrl());
    }

    public static final List<Library> NECESSARY_LIBRARIES = List.of(
            new Library("org.jetbrains.kotlin", "kotlin-stdlib", "1.9.23")
    );

    public String getUrl() {
        return group.replace(".", "/")
                + "/" + name + "/" + version + "/"
                + getFileName();
    }

    public String getFileName() {
        return name + "-" + version + ".jar";
    }

    public String getDisplayName() {
        return group + ":" + name + ":" + version;
    }
}

File: velocity/src/main/kotlin/moe/caa/multilogin/velocity/loader/library/Library.kt
package moe.caa.multilogin.velocity.loader.library

import moe.caa.multilogin.velocity.util.md5Digest
import java.io.File

data class Library(
    val group: String,
    val name: String,
    val version: String,
    val md5: ByteArray
) {
    private val fileName = "$name-$version.jar"
    private val url = "${group.replace('.', '/')}/$name/$version/$fileName"

    fun getFile(folder: File): File {
        return File(folder, url)
    }


    fun md5Check(libraryFile: File): Boolean{
        return libraryFile.md5Digest().contentEquals(md5)
    }
}
